File: routeMap/app/src/main/java/ru/android/zheka/fragment/Map.java
package ru.android.zheka.fragment;

import javax.inject.Inject;

import ru.android.zheka.coreUI.AbstractFragment;
import ru.android.zheka.gmapexample1.R;
import ru.android.zheka.gmapexample1.databinding.MapFragmentBinding;
import ru.android.zheka.vm.IMapVM;

public class Map extends AbstractFragment <MapFragmentBinding> {
    @Inject
    IMapVM viewModel;

    @Override
    protected int getLayoutId() {
        return R.layout.map_fragment;
    }

    @Override
    protected void initComponent() {

    }

    @Override
    protected void onInitBinding(MapFragmentBinding binding) {
        binding.setVm (viewModel);
    }

    @Override
    protected void onResumeBinding(MapFragmentBinding binding) {
        viewModel.onResume ();
    }

    @Override
    protected void onDestroyBinding(MapFragmentBinding binding) {
        viewModel.onDestroy ();
    }
}

File: routeMap/app/src/main/java/ru/android/zheka/fragment/Map.kt
package ru.android.zheka.fragment

import kotlinx.android.synthetic.main.home_fragment.view.*
import ru.android.zheka.coreUI.AbstractFragment
import ru.android.zheka.coreUI.IActivity
import ru.android.zheka.gmapexample1.R
import ru.android.zheka.gmapexample1.databinding.MapFragmentBinding
import ru.android.zheka.vm.IMapVM
import javax.inject.Inject

class Map : AbstractFragment<MapFragmentBinding>(), IActivity {
    @JvmField
    @Inject
    var viewModel: IMapVM? = null
    override
    val layoutId = R.layout.map_fragment

    override fun initComponent() {}
    override fun onInitBinding(binding: MapFragmentBinding) {
        binding.vm = viewModel
    }

    override fun onResumeBinding(binding: MapFragmentBinding) {
        viewModel!!.onResume()
    }

    override fun onDestroyBinding(binding: MapFragmentBinding) {
        viewModel!!.onDestroy()
    }

    override fun initAdapter(binding: MapFragmentBinding): MapFragmentBinding {
        val scale = context?.resources?.displayMetrics?.density
        var value: Int = ((scale ?: 0F).times(Geo.sp.toFloat())).toInt()
        value = if (value == 0) 230 else value
        binding.root.baseScanBarcode_background?.layoutParams?.height = value
        return binding
    }
}